lib/rexml/element.rb

```diff
@@ -589,10 +589,12 @@ def prefixes
     #   d.elements['//c'].namespaces # => {"x"=>"1", "y"=>"2", "z"=>"3"}
     #
     def namespaces
-      namespaces = {}
-      namespaces = parent.namespaces if parent
-      namespaces = namespaces.merge( attributes.namespaces )
-      return namespaces
+      namespaces_cache = root.document&.namespaces_cache
```

Comment:
Do we need `root` here?

```suggestion
      namespaces_cache = document&.namespaces_cache
```